Clarified Points and Scatter notebooks

jbednar opened the pull request:
Fixes issue 1 from https://github.com/holoviz/holoviews/issues/4760 . Still need to address issue 2.

philippjfr commented on examples/reference/elements/bokeh/Points.ipynb:
```diff
@@ -147,10 +147,9 @@
-    "This difference means that ``Points`` naturally combine elements that express independent variables in two-dimensional space, for instance [``Raster``](./Raster.ipynb) types such as [``Image``](./Image.ipynb). Similarly, ``Scatter`` expresses a dependent relationship in two-dimensions and combine naturally with ``Chart`` types such as [``Curve``](./Curve.ipynb).\n",
+    "**Note**: Although the  ``Scatter`` element is superficially similar to the [``Points``](./Points.ipynb) element (they can generate plots that look identical), the two element types are semantically quite different. The fundamental difference is that [Scatter](./Scatter.ipynb) is used to visualize data where the *y* variable is *dependent*, unlike ``Points``. This semantic difference also ex ...
     "\n",
+    "This difference means that ``Points`` elements can easily be overlaid with other elements that express independent variables in two-dimensional space, such as [``Raster``](./Raster.ipynb) types like [``Image``](./Image.ipynb). Conversely, ``Scatter`` expresses a dependent relationship between *x* and *y* and thus overlays easily with ``Chart`` types such as [``Curve``](./Curve.ipynb).\n",
```
While this is true conceptually people might be confused by this since scatter overlays on easter/image just fine.

jbednar replied:
Yeah; maybe you can think of a better way to say that. "most naturally"?

philippjfr replied:
"Most naturally" works for me, "easily" was my main objection.

jbednar replied:
I'll update to "most naturally.

philippjfr commented:
Thanks, one comment which you can ignore.

jbednar commented:
For point 2, my guess is that all the examples used to say `.hist()` and thus , but someone (potentially me!) noticed that the histograms didn't match between Points and Scatter, so they made Scatter use `.hist('z')` and thus made the histograms match.  I've reverted that, and now .hist() selects different dimensions on the two object types (always the first value dimension, i.e. 'y' on Scatter and 'z' on Points) .

I also removed any mention of .hist() on the Plotly versions, because those examples don't actually use histograms, presumably because AdjointLayouts are not supported for that backend.

jbednar commented on examples/reference/elements/bokeh/Points.ipynb:
```diff
@@ -70,7 +70,7 @@
    "metadata": {},
    "source": [
-    "Although the simplest ``Points`` element simply marks positions in a two-dimensional space without any associated value, other value dimensions (``vdims``) are also supported. Here is an example with two additional quantities for each point, declared as the ``vdims`` ``'z'`` and ``'size'`` visualized as the color and size of the dots, respectively:"
+    "Although the simplest ``Points`` element simply marks positions in a two-dimensional space without any associated value, value dimensions (``vdims``)  are also supported. Here is an example with two additional quantities for each point, declared as the ``vdims``s ``z`` and ``size`` (visualized as the color and size of the dots, respectively):"
```
```suggestion
    "Although the simplest ``Points`` element simply marks positions in a two-dimensional space without any associated value, value dimensions (``vdims``) are also supported. Here is an example with two additional quantities for each point, declared as the ``vdims``s ``z`` and ``size`` (visualized as the color and size of the dots, respectively):"
```

jbednar commented on examples/reference/elements/plotly/Points.ipynb:
```diff
@@ -66,7 +66,7 @@
    "metadata": {},
    "source": [
-    "Although the simplest ``Points`` element simply mark positions in a two-dimensional space without any associated value this doesn't mean value dimensions aren't supported. Here is an example with two additional quantities for each point, declared as the ``value_dimension``s *z* and &alpha; visualized as the color and size of the dots, respectively:"
+    "Although the simplest ``Points`` element simply marks positions in a two-dimensional space without any associated value, value dimensions (``vdims``)  are also supported. Here is an example with two additional quantities for each point, declared as the ``vdims``s ``z`` and ``size`` (visualized as the color and size of the dots, respectively):"
```
```suggestion
    "Although the simplest ``Points`` element simply marks positions in a two-dimensional space without any associated value, value dimensions (``vdims``) are also supported. Here is an example with two additional quantities for each point, declared as the ``vdims``s ``z`` and ``size`` (visualized as the color and size of the dots, respectively):"
```

jbednar commented:
Ok, this is ready for review and merge. It makes me go blind to try to read through these six nearly identical but subtly different notebooks, but I *think* each of them has the correct changes now.

jbednar commented:
I actually tried out Plotly with .hist(), and I think that was missing before because used to fail but it's working fine now, so I made the Plotly examples include .hist() to match the Bokeh and Matplotlib ones. I also removed other seemingly unnecessary differences like colormaps and slice ranges, so that all the notebooks now show similar stuff.

Oddly, the Bokeh Scatter notebook is failing with `KeyError: "One or more dimensions in the expression dim('z') could not resolve on ':Histogram   [y]   (y_count)'. Ensure all dimensions referenced by the expression are present on the supplied object."` , for the same code that works on the Matplotlib Scatter notebook. @philippjfr, can you take a look? Either I'm confused or there is a bug.

philippjfr commented:
@jbednar It was actually your fault 😄 The problem is that you dropped the `'z'` from `.hist` call, so the histogram was on the wrong dimension. The matplotlib version that "worked" was still coloring by the wrong dimension so wasn't really any good.

jbednar commented:
That was deliberate; it was _supposed_ to be a histogram on 'y'. Reverting that is not a fix; how can you get a histogram on 'y'?
